### PR TITLE
Location: subclasses should have different hashes

### PR DESCRIPTION
**Pull request by dhalperi**
This is basically similar to batfish/batfish#3855: hash codes should not be the same even if the two fields are.

### CODE REVIEW

**Review by anothermattbrown — COMMENTED** [review]
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/batfish/batfish/6266)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @corinaminer)

___
*[projects/batfish-common-protocol/src/main/java/org/batfish/specifier/InterfaceLinkLocation.java, line 57 at r1](https://reviewable.io/reviews/batfish/batfish/6266#-MIZyKVT8JOGvDQhRHbM:-MIZyKVT8JOGvDQhRHbN:b-pu3f0) ([raw file](https://github.com/batfish/batfish/blob/9ed00f878d7793d8d55c6f4bb11dba85bda3eb25/projects/batfish-common-protocol/src/main/java/org/batfish/specifier/InterfaceLinkLocation.java#L57)):*
> ```Java
>   @Override
>   public int hashCode() {
>     return 31 * 31 * getClass().hashCode() + 31 * _interfaceName.hashCode() + _nodeName.hashCode();
> ```

why not just add `getClass` to the `Objects.hash` call?

**Comment by dhalperi** [comment]
___
*[projects/batfish-common-protocol/src/main/java/org/batfish/specifier/InterfaceLinkLocation.java, line 57 at r1](https://reviewable.io/reviews/batfish/batfish/6266#-MIZyKVT8JOGvDQhRHbM:-MI_G1CD5KqP7PrFHhSS:bfojnmm) ([raw file](https://github.com/batfish/batfish/blob/9ed00f878d7793d8d55c6f4bb11dba85bda3eb25/projects/batfish-common-protocol/src/main/java/org/batfish/specifier/InterfaceLinkLocation.java#L57)):*
<details><summary><i>Previously, anothermattbrown (Matt Brown) wrote…</i></summary><blockquote>

why not just add `getClass` to the `Objects.hash` call?
</blockquote></details>

tl;dr: Objects.hash creates an array, which is unnecessary churn for code that may be hot.
